controller/metrics/wrap.go

```diff
@@ -0,0 +1,164 @@
+package metrics
```

Comment:
This package was largely stolen from argo-rollouts, but has a bunch of bug fixes. I can also move this to argo/pkg instead.

Reply:
@dthomson25 shall I do this?